solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -913,7 +924,9 @@ private void init() {
         ElectionContext context = new OverseerElectionContext(zkClient,
             overseer, getNodeName());
         overseerElector.setup(context);
-        overseerElector.joinElection(context, false);
+        if (cc.nodeRoles.isOverseerAllowedOrPreferred()) {
```

Comment:
Can we expand this conditional to avoid getting the context above?

Reply:
Tried doing so, but ran into test failures. Maybe @noblepaul might know why this doesn't work?